src/rust/multichan/template/src/template.rs

```diff
@@ -0,0 +1,392 @@
+			.text()
+			.read();
+		let status = if status_str.contains("перевод продолжается") {
+			MangaStatus::Ongoing
+		} else if status_str.contains("перевод завершен") {
+			MangaStatus::Completed
+		} else {
+			MangaStatus::Unknown
+		};
+		let (nsfw, viewer) = category_parser(&categories);
+		Ok(Manga {
+			id,
+			cover,
+			title,
+			author: String::from(author.trim()),
+			artist: String::new(),
+			description,
+			url,
+			categories,
+			status,
+			nsfw,
+			viewer,
+		})
+	}
+
+	pub fn get_chapter_list(&self, id: String) -> Result<Vec<Chapter>> {
+		cache_manga_page(&format!("{}{id}", self.base_url));
+		let html = Node::new(unsafe { &CACHED_MANGA.clone().unwrap() });
+		let manga_title = html.select("a.title_top_a").text().read();
+		let scanlator = html
+			.select("a[href*=translation][title]")
+			.array()
+			.filter_map(|elem| {
+				let text = elem.as_node().text().read();
+				if text.is_empty() {
+					None
+				} else {
+					Some(text)
+				}
+			})
+			.collect::<Vec<_>>()
+			.join(", ");
+		let node = html.select("table.table_cha tr[class*=zaliv]");
+		Ok(node
+			.array()
+			.map(|elem| {
+				let chapter_node = elem.as_node();
+				let id = chapter_node.select("a").attr("href").read();
+				let url = format!("{}{id}", self.base_url);
+				let date_updated = chapter_node
+					.select("div.date")
+					.text()
+					.0
+					.as_date("yyyy-MM-dd", None, None)
+					.unwrap_or(-1.0);
+				let mut title = chapter_node.select("a").text().read();
+				let (volume, chapter) = (self.vol_chap_parser)(manga_title.clone(), title.clone());
+				if chapter >= 0.0 {
+					let splitter = format!(" {}", chapter);
+					if title.contains(&splitter) {
+						let split = title.splitn(2, &splitter).collect::<Vec<&str>>();
+						title = String::from(split[1]).replacen(
+							|char| char == ':' || char == '-',
+							"",
+							1,
+						);
+					}
+				}
+				Chapter {
+					id,
+					title: String::from(title.trim()),
+					volume,
+					chapter,
+					date_updated,
+					scanlator: scanlator.clone(),
+					url,
+					lang: String::from("ru"),
+				}
+			})
+			.collect::<Vec<_>>())
+	}
+
+	pub fn get_page_list(&self, id: String) -> Result<Vec<Page>> {
+		let url = if id.starts_with("http") {
+			// exhentai-dono.me
+			format!("{id}&development_access=true")
+		} else {
+			format!("{}{id}", self.base_url)
+		};
+		let html = Request::new(&url, HttpMethod::Get).html().html().read();
+		let (begin, end) = if let Some(begin_) = html.find("fullimg\":[") {
+			// manga-chan and yaoi-chan
+			let begin = begin_ + 10;
+			let end = html[begin..].find(",]").map(|i| i + begin).unwrap_or(0);
+			(begin, end)
+		} else if let Some(begin_) = html.find("fullimg\": [") {
+			// hentai-chan
+			println!("hentaichan");
+			let begin = begin_ + 11;
+			let end = html[begin..].find(']').map(|i| i + begin).unwrap_or(0);
+			(begin, end)
+		} else {
+			(0, 2)
+		};
+		println!("{}", &html[begin..end]);
```

Comment:
Probably best not to have printing in public sources unless necessary.
```suggestion
		} else if let Some(begin_) = html.find("fullimg\": [") {
			// hentai-chan
			let begin = begin_ + 11;
			let end = html[begin..].find(']').map(|i| i + begin).unwrap_or(0);
			(begin, end)
		} else {
			(0, 2)
		};
```